docs/en/ingest-management/fleet/add-fleet-server-on-prem.asciidoc

```diff
@@ -0,0 +1,154 @@
+[[add-fleet-server-on-prem]]
+= Deploy on-premises and self-managed
+
+To deploy a self-managed {fleet-server}, you need to: 
+
+* Satisfy all <<add-fleet-server-on-prem-compatibility,compatibility requirements and prerequisites>>
+* Add hosts including <<add-fleet-server-on-prem-hosts,{fleet-server} hosts and {es} hosts>>
+* Install an {agent} and enroll it in an agent policy containing the {fleet-server} integration to <<add-fleet-server-on-prem-add-server,Add {fleet-server}>>
+
+NOTE: You can install only a single {agent} per host, which means you cannot run
+{fleet-server} and another {agent} on the same host unless you deploy a
+containerized {fleet-server}.
+
+[discrete]
+[[add-fleet-server-on-prem-compatibility]]
+== Compatibility and prerequisites
+
+{fleet-server} is compatible with the following Elastic products:
+
+* {stack} 7.13 or later.
+** For version compatibility: {es} >= {fleet-server} >= {agent} (except for
+bugfix releases)
+** {kib} should be on the same minor version as {es}.
+
+* {ece} 2.9 or later
++
+--
+** Requires additional wildcard domains and certificates (which normally only
+cover `*.cname`, not `*.*.cname`). This enables us to provide the URL for
+{fleet-server} of `https://.fleet.`.
+** The deployment template must contain an {integrations-server} node.
+--
++
+For more information about hosting {fleet-server} on {ece}, refer to
+{ece-ref}/ece-manage-integrations-server.html[Manage your {integrations-server}].
+
+[discrete]
+[[add-fleet-server-on-prem-hosts]]
+== Add hosts
```

Comment:
I split the steps into two sections:

* Add hosts
* Add Fleet Server

I did this in part because these instructions cover two distinct parts of the UI, but also it felt more digestible to me to break it into two concepts (but that's subjective!).

Reply:
sounds good to me